### PR TITLE
Add ts_simple_recompile_ddl

### PR DESCRIPTION
**Pull request by javajolt**
This test verifies that tables with mismatched DDL versions or old (i.e. missing) DDL compiler versions are actually recompiled.  The greatest difficulty in verifying this test was reading the results from the DETS table.  It turns out the trick is that the default DETS write-to-disk interval is 3 minutes, to by adding a manual `dets:sync` with each write, the test became much simpler.

Requires https://github.com/basho/riak_kv/pull/1377 and https://github.com/basho/riak_ql/pull/115

### CODE REVIEW

File: tests/ts_simple_recompile_ddl.erl [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+-define(TABLE, ?MODULE).
+-define(KV_MODULE, riak_kv_compile_tab).
+
+confirm() ->
+    {Cluster, _Conn} = ts_util:cluster_and_connect(single),
+    Node = hd(Cluster),
+    lists:foreach(
+        fun(Table) ->
+            DDL = create_table_sql(Table),
+            ts_util:create_and_activate_bucket_type(Cluster, DDL, Table)
```
Comment (hmmr) [inline tests/ts_simple_recompile_ddl.erl]: In the new test modules, why not use the direct "CREATE TABLE" query rather than two separate, `riak-admin` based commands?

Comment (javajolt) [inline tests/ts_simple_recompile_ddl.erl]: Great question. Perhaps because it wasn't there originally when the testing infrastructure was built?  It would be an easy change, but let's do that in a separate PR.

Comment (hmmr) [inline tests/ts_simple_recompile_ddl.erl]: Agree. Possibly, also consider adding a subtest to ts_cluster_create_table_pass to check the equivalence of creating a table via query vs creating table via riak-admin.

File: tests/ts_simple_recompile_ddl.erl [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+    open_dets(),
+    Pid = spawn_link(fun() -> ok end),
+    Pid2 = spawn_link(fun() -> ok end),
+    Pid3 = spawn_link(fun() -> ok end),
+    Table1DDL = sql_to_ddl(create_table_sql("Table1")),
+    Table2DDL = sql_to_ddl(create_table_sql("Table2")),
+    Table3DDL = sql_to_ddl(create_table_sql("Table3")),
+    ok = dets:insert(?TABLE, {<<"Table1">>, Table1DDL, Pid, compiled}),
+    ok = dets:insert(?TABLE, {<<"Table2">>, 1, Table2DDL, Pid2, compiled}),
+    ok = dets:insert(?TABLE, {<<"Table3">>, 1, Table3DDL, Pid3, compiling}),
```
Comment (hmmr) [inline tests/ts_simple_recompile_ddl.erl]: That `compiling` atom you are inserting last, isn't it going to make the `verify_resulting_dets_entries` fail?

Generally, some comments here would help reviewers to understand the underlying logic of the test. As it is, in `verify_resulting_dets_entries` you are reading the entries you have just inserted (in `simulate_old_dets_entries`), which is bound to succeed.

Aha, got it. The `'compiling'` will be turned to `'compiled'` in the process of node going up. I'm smart, no?

**Comment by hmmr** [comment]
+1 1389ab6

File: tests/ts_simple_recompile_ddl.erl (hunk 2 above)
Comment (javajolt) [inline tests/ts_simple_recompile_ddl.erl]: You are, indeed, smart.  Let me add some comments on the 3 cases being tested.

**Comment by hmmr** [comment]
+1 3240f09

**Comment by macintux** [comment]
:+1:
